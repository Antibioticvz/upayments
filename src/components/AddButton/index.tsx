import { FC } from 'react'
import { useNavigate } from 'react-router-dom'

import { ROUTES } from 'constants/routes'

import './styles.css'

export const AddButton: FC = () => {
  const navigate = useNavigate()

  return (
    <button
      type="button"
      data-mdb-ripple="true"
      data-mdb-ripple-color="light"
      className="inline-block p-3 bg-black text-white font-medium text-xs leading-tight uppercase rounded-full shadow-md hover:bg-green-700 hover:shadow-lg focus:bg-green-700 focus:shadow-lg focus:outline-none focus:ring-0 active:bg-green-800 active:shadow-lg transition duration-150 ease-in-out bottom-5 right-5"
      id="btn-back-to-top"
      onClick={() => navigate(ROUTES.CREATE)}
    >
      <svg
        aria-hidden="true"
        focusable="false"
        data-prefix="fas"
        className="w-7 h-7"
        role="img"
        xmlns="http://www.w3.org/2000/svg"
        viewBox="0 0 330 350"
      >
        <path
          fill="currentColor"
          d="M349.03,141.226v66.579c0,5.012-4.061,9.079-9.079,9.079H216.884v123.067c0,5.019-4.067,9.079-9.079,9.079h-66.579   c-5.009,0-9.079-4.061-9.079-9.079V216.884H9.079c-5.016,0-9.079-4.067-9.079-9.079v-66.579c0-5.013,4.063-9.079,9.079-9.079   h123.068V9.079c0-5.018,4.069-9.079,9.079-9.079h66.579c5.012,0,9.079,4.061,9.079,9.079v123.068h123.067   C344.97,132.147,349.03,136.213,349.03,141.226z"
        />
      </svg>
    </button>
  )
}
