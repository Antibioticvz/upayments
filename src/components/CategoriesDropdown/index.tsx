import { FC } from 'react'
import { Listbox, Transition } from '@headlessui/react'
import { ICategories } from 'requests'

interface ICategoriesDropdown {
  options: ICategories[]
  selectedCategories: string[]
  // eslint-disable-next-line no-unused-vars
  handleSelect: (value: string) => void
}

export const CategoriesDropdown: FC<ICategoriesDropdown> = ({
  options,
  selectedCategories,
  handleSelect,
}) => (
  <div className="flex justify-between m-10 max-w-full ">
    <div className="w-3/12 lg:w-6/12 md:w-4/12 sm:w-3/12 relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left shadow-md">
      <p className="overflow-hidden truncate ">
        {selectedCategories.length > 0 ? (
          <> {selectedCategories.join(', ')}</>
        ) : (
          'All categories'
        )}
      </p>
    </div>

    <div className="w-8/12 lg:w-4/12 md:w-6/12 sm:w-8/12">
      <Listbox
        as="div"
        className="space-y-1"
        value={selectedCategories}
        // Its a bug of Listbox. In fact we have  value: string not string[]
        onChange={(value: any) => handleSelect(value)}
      >
        {({ open }) => (
          <div className="relative">
            <span className="inline-block w-full rounded-md shadow-sm ">
              <Listbox.Button className="cursor-default relative w-full rounded-md border border-gray-300 bg-white shadow-md pl-3 pr-10 py-2 text-left  transition ease-in-out duration-150">
                <span className="block truncate">Categories</span>
                <span className="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none">
                  <svg
                    className="ml-2 w-4 h-4"
                    fill="none"
                    stroke="currentColor"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      d="M19 9l-7 7-7-7"
                    />
                  </svg>
                </span>
              </Listbox.Button>
            </span>

            <Transition
              unmount={false}
              show={open}
              leave="transition ease-in duration-100"
              leaveFrom="opacity-100"
              leaveTo="opacity-0"
              className="absolute mt-1 w-full rounded-md bg-white shadow-lg"
            >
              <Listbox.Options
                static
                className="max-h-60 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
              >
                {options.map(({ id, name }) => {
                  const selected = !!selectedCategories.find(el => el === name)

                  return (
                    <Listbox.Option key={id} value={name}>
                      {({ active }) => (
                        <div
                          className={`${
                            active ? 'text-white bg-blue-600' : 'text-gray-900'
                          } cursor-default select-none relative py-2 pl-8 pr-4`}
                        >
                          <span
                            className={`${
                              selected ? 'font-semibold' : 'font-normal'
                            } block truncate`}
                          >
                            {name}
                          </span>
                          {selected && (
                            <span
                              className={`${
                                active ? 'text-white' : 'text-blue-600'
                              } absolute inset-y-0 left-0 flex items-center pl-1.5`}
                            >
                              <svg
                                className="h-5 w-5"
                                xmlns="http://www.w3.org/2000/svg"
                                viewBox="0 0 20 20"
                                fill="currentColor"
                              >
                                <path
                                  fillRule="evenodd"
                                  d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                                  clipRule="evenodd"
                                />
                              </svg>
                            </span>
                          )}
                        </div>
                      )}
                    </Listbox.Option>
                  )
                })}
              </Listbox.Options>
            </Transition>
          </div>
        )}
      </Listbox>
    </div>
  </div>
)
