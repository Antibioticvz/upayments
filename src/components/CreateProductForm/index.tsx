import { FC } from 'react'

export const CreateProductForm: FC = () => <div>CreateProductForm</div>
