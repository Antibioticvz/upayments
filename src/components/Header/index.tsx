import { FC } from 'react'

export const Header: FC = () => (
  <div className="flex justify-between m-10 max-w-full rounded-xl bg-white border-2 border-gray-300 shadow-md">
    <p className="text-xl font-semibold m-2">UPaymentStore</p>
    <p className="text-xl  font-semibold m-2">Register</p>
  </div>
)
