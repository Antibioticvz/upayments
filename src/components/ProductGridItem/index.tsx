import { FC, useCallback } from 'react'
import { useNavigate } from 'react-router-dom'

import { ROUTES_HREF } from 'constants/routes'
import { IProduct } from 'requests'

export const ProductGridItem: FC<IProduct> = ({ id, name, avatar, price }) => {
  const navigate = useNavigate()

  const getRoundedPrice = useCallback(
    () =>
      Number(
        `${price}`.replace(',', '.').replace(/[^0-9&.]/g, '') || 0,
      ).toFixed(2),
    [price],
  )

  return (
    <div
      role="button"
      tabIndex={0}
      className="flex justify-between flex-col"
      onKeyPress={e => e.preventDefault()}
      onClick={() => navigate(`${ROUTES_HREF.PRODUCT}/${id}`)}
    >
      <img
        src={avatar}
        className="max-w-full h-auto rounded-2xl bg-white border-2 border-gray-300"
        alt={name}
      />
      <div>
        <p className="text-xl text-left mt-5 mb-2">{name}</p>
        <p className="text-xl text-center">$ {getRoundedPrice()}</p>
      </div>
    </div>
  )
}
