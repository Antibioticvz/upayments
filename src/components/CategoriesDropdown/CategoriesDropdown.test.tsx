import { render } from '@testing-library/react'

import { CategoriesDropdown } from 'components/CategoriesDropdown'

const options = [
  {
    'createdAt': '2022-04-14T18:27:19.838Z',
    'name': 'Electronic',
    'id': '1',
  },
  {
    'createdAt': '2022-04-14T18:27:19.838Z',
    'name': 'Furnitures',
    'id': '2',
  },
  {
    'createdAt': '2022-04-14T18:27:19.838Z',
    'name': 'Clothing',
    'id': '3',
  },
  {
    'createdAt': '2022-04-14T18:27:19.838Z',
    'name': 'Accessories',
    'id': '4',
  },
]

describe('CategoriesDropdown component', () => {
  test('renders CategoriesDropdown', () => {
    const { asFragment } = render(
      <CategoriesDropdown
        options={options}
        selectedCategories={[]}
        handleSelect={value => console.log(value)}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders CategoriesDropdown / One selected', () => {
    const { asFragment } = render(
      <CategoriesDropdown
        options={options}
        selectedCategories={['Accessories']}
        handleSelect={value => console.log(value)}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders CategoriesDropdown / All selected', () => {
    const { asFragment } = render(
      <CategoriesDropdown
        options={options}
        selectedCategories={[
          'Electronic',
          'Furnitures',
          'Clothing',
          'Accessories',
        ]}
        handleSelect={value => console.log(value)}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
