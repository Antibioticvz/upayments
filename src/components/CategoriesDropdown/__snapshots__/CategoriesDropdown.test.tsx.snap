// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesDropdown component renders CategoriesDropdown / All selected 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between m-10 max-w-full "
  >
    <div
      class="w-3/12 lg:w-6/12 md:w-4/12 sm:w-3/12 relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left shadow-md"
    >
      <p
        class="overflow-hidden truncate "
      >
         Electronic, Furnitures, Clothing, Accessories
      </p>
    </div>
    <div
      class="w-8/12 lg:w-4/12 md:w-6/12 sm:w-8/12"
    >
      <div
        class="space-y-1"
      >
        <div
          class="relative"
        >
          <span
            class="inline-block w-full rounded-md shadow-sm "
          >
            <button
              aria-controls="headlessui-listbox-options-17"
              aria-expanded="false"
              aria-haspopup="true"
              class="cursor-default relative w-full rounded-md border border-gray-300 bg-white shadow-md pl-3 pr-10 py-2 text-left  transition ease-in-out duration-150"
              id="headlessui-listbox-button-15"
              type="button"
            >
              <span
                class="block truncate"
              >
                Categories
              </span>
              <span
                class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
              >
                <svg
                  class="ml-2 w-4 h-4"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 9l-7 7-7-7"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </button>
          </span>
          <div
            class="absolute mt-1 w-full rounded-md bg-white shadow-lg transition ease-in duration-100 opacity-100"
            hidden=""
            style="display: none;"
          >
            <ul
              aria-labelledby="headlessui-listbox-button-15"
              aria-orientation="vertical"
              class="max-h-60 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
              id="headlessui-listbox-options-17"
              role="listbox"
              tabindex="0"
            >
              <li
                id="headlessui-listbox-option-18"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-semibold block truncate"
                  >
                    Electronic
                  </span>
                  <span
                    class="text-blue-600 absolute inset-y-0 left-0 flex items-center pl-1.5"
                  >
                    <svg
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-19"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-semibold block truncate"
                  >
                    Furnitures
                  </span>
                  <span
                    class="text-blue-600 absolute inset-y-0 left-0 flex items-center pl-1.5"
                  >
                    <svg
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-20"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-semibold block truncate"
                  >
                    Clothing
                  </span>
                  <span
                    class="text-blue-600 absolute inset-y-0 left-0 flex items-center pl-1.5"
                  >
                    <svg
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-21"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-semibold block truncate"
                  >
                    Accessories
                  </span>
                  <span
                    class="text-blue-600 absolute inset-y-0 left-0 flex items-center pl-1.5"
                  >
                    <svg
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CategoriesDropdown component renders CategoriesDropdown / One selected 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between m-10 max-w-full "
  >
    <div
      class="w-3/12 lg:w-6/12 md:w-4/12 sm:w-3/12 relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left shadow-md"
    >
      <p
        class="overflow-hidden truncate "
      >
         Accessories
      </p>
    </div>
    <div
      class="w-8/12 lg:w-4/12 md:w-6/12 sm:w-8/12"
    >
      <div
        class="space-y-1"
      >
        <div
          class="relative"
        >
          <span
            class="inline-block w-full rounded-md shadow-sm "
          >
            <button
              aria-controls="headlessui-listbox-options-10"
              aria-expanded="false"
              aria-haspopup="true"
              class="cursor-default relative w-full rounded-md border border-gray-300 bg-white shadow-md pl-3 pr-10 py-2 text-left  transition ease-in-out duration-150"
              id="headlessui-listbox-button-8"
              type="button"
            >
              <span
                class="block truncate"
              >
                Categories
              </span>
              <span
                class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
              >
                <svg
                  class="ml-2 w-4 h-4"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 9l-7 7-7-7"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </button>
          </span>
          <div
            class="absolute mt-1 w-full rounded-md bg-white shadow-lg transition ease-in duration-100 opacity-100"
            hidden=""
            style="display: none;"
          >
            <ul
              aria-labelledby="headlessui-listbox-button-8"
              aria-orientation="vertical"
              class="max-h-60 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
              id="headlessui-listbox-options-10"
              role="listbox"
              tabindex="0"
            >
              <li
                id="headlessui-listbox-option-11"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Electronic
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-12"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Furnitures
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-13"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Clothing
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-14"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-semibold block truncate"
                  >
                    Accessories
                  </span>
                  <span
                    class="text-blue-600 absolute inset-y-0 left-0 flex items-center pl-1.5"
                  >
                    <svg
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CategoriesDropdown component renders CategoriesDropdown 1`] = `
<DocumentFragment>
  <div
    class="flex justify-between m-10 max-w-full "
  >
    <div
      class="w-3/12 lg:w-6/12 md:w-4/12 sm:w-3/12 relative w-full rounded-md border border-gray-300 bg-white pl-3 pr-10 py-2 text-left shadow-md"
    >
      <p
        class="overflow-hidden truncate "
      >
        All categories
      </p>
    </div>
    <div
      class="w-8/12 lg:w-4/12 md:w-6/12 sm:w-8/12"
    >
      <div
        class="space-y-1"
      >
        <div
          class="relative"
        >
          <span
            class="inline-block w-full rounded-md shadow-sm "
          >
            <button
              aria-controls="headlessui-listbox-options-6"
              aria-expanded="false"
              aria-haspopup="true"
              class="cursor-default relative w-full rounded-md border border-gray-300 bg-white shadow-md pl-3 pr-10 py-2 text-left  transition ease-in-out duration-150"
              id="headlessui-listbox-button-1"
              type="button"
            >
              <span
                class="block truncate"
              >
                Categories
              </span>
              <span
                class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none"
              >
                <svg
                  class="ml-2 w-4 h-4"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M19 9l-7 7-7-7"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </span>
            </button>
          </span>
          <div
            class="absolute mt-1 w-full rounded-md bg-white shadow-lg transition ease-in duration-100 opacity-100"
            hidden=""
            style="display: none;"
          >
            <ul
              aria-labelledby="headlessui-listbox-button-1"
              aria-orientation="vertical"
              class="max-h-60 rounded-md py-1 text-base leading-6 shadow-xs overflow-auto focus:outline-none sm:text-sm sm:leading-5"
              id="headlessui-listbox-options-6"
              role="listbox"
              tabindex="0"
            >
              <li
                id="headlessui-listbox-option-2"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Electronic
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-3"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Furnitures
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-4"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Clothing
                  </span>
                </div>
              </li>
              <li
                id="headlessui-listbox-option-5"
                role="option"
                tabindex="-1"
              >
                <div
                  class="text-gray-900 cursor-default select-none relative py-2 pl-8 pr-4"
                >
                  <span
                    class="font-normal block truncate"
                  >
                    Accessories
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
